packages/jest-haste-map/src/worker.js

```diff
@@ -22,25 +21,7 @@ const PACKAGE_JSON = path.sep + 'package' + JSON_EXTENSION;
 let hasteImpl: ?HasteImpl = null;
 let hasteImplModulePath: ?string = null;
 
-const formatError = (error: string | Error): SerializableError => {
```

Comment:
Very happy this is gone!

Reply:
@mjesun can you make sure that error handling still works correctly? For example in `jest-runner` (separate PR), when a user throws a string, it will work and report the proper error object back. In `reportError` in `jest-worker`'s `child.js` I cannot see where this is handled.

Reply:
Jest workers rejects parent promises that represent a worker call by using the exact same value that you used for throwing. If you do `throw 'foo'`, the overlying promise will be rejected with `'foo'`, so it is fully transparent for the `jest-worker` user.

I might have gone wrong expecting `jest` core to be able to handle non-`Error` exceptions, since the actual code only handles `string`, while you can throw anything like `throw 3` or `throw true`. If that's not the case, then I will have to add the wrapper, but that would be on the parent's code and not on the child. I removed it because it seemed more like something ad-hoc for `worker-farm` rather than for Jest.

Reply:
Yeah, the parents normally expect real error objects in Jest, which is why this code was there. It's probably less important for `jest-haste-map` than it is for `jest-runner`, though.

Reply:
I'm wondering how this deals right now with things like `throw {}`, `throw 3` or `throw false`. Looks like the actual implementation is flaky either way? Happy to improve it generically, though 🙂. What about doing something like:

```javascript
if (!error instanceof Error) {
  error = new Error(error);
}
```

Reply:
I'd rather detect the shape than doing instanceof. It's not guaranteed that the thrown error object actually descends from the Error class. Especially with `vm` classes involved, you are very likely dealing with multiple different instances of the Error class and then instances of those Error classes.

Reply:
`jest-worker` will deserialize the object to an `Error` instance if it detects that the thrown error on the child is an instance of `Error`. Under this circumstances, I'd say that the `instanceof` check is valid? (I'm not 100% sure)

Reply:
What I'm saying is that you cannot rely on errors thrown in the child to be an instance of any one Error class in particular, and at any given time there may be dozens of instances of Error and people may throw stuff that aren't errors. `formatError` previously was meant to deal with these cases. As long as it doesn't change how that works, I'm fine with anything.

Reply:
I see; OK, let me do some extra testing and I'll come up with something.

Reply:
So here's what I did: I used duck typing to detect errors; if it's not, then I just wrap it as an error, and the code continues. The duck typing condition I'm using is:

```javascript
typeof error !== 'object' || !error.message || !error.stack
```

I already pushed the change with this.